src/operator/cudnn_ops.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file  cudnn_ops.h
+ * \brief cuDNN v8 ops
+ */
+#ifndef MXNET_OPERATOR_CUDNN_OPS_H_
+#define MXNET_OPERATOR_CUDNN_OPS_H_
+
+#include <mxnet/base.h>
+#if MXNET_USE_CUDNN == 1
+
+#include <mxnet/op_attr_types.h>
+
+#include <mutex>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "nn/convolution-inl.h"
+#include "nn/deconvolution-inl.h"
+
+#include "../common/cuda/cudnn_cxx.h"
+
+namespace mxnet {
+namespace tuple_util {
```

Comment:
Why is it here and not in the cudnn_cxx.h?

Reply:
Just by default, I'm trying to make stuff as local / private as possible? For now, just considering it as an implementation detail of cudnn_ops. Would have put it into cudnn_ops.cc, had it not been used in cudnn_ops.h.

On the other hand, there's nothing cuDNN related here. So, if we do want to make it generic and reusable, the proper place would probably be to put it into dmlc, somewhere close to where dmlc::HashCombine is defined? I can prepare a separate PR to move it from here to there.

Reply:
I think that's fine. dmlc-core is a separate repo so that is probably unnecessary to go there.